### DISCUSSION TITLE
Nested writes and two relations between two the same models

### DISCUSSION THREAD

**Discussion by yarikos** [prisma/prisma#3483] [Q&A] [answered]
Let's consider an example from
https://www.prisma.io/docs/reference/tools-and-interfaces/prisma-schema/relations#disambiguating-relations

```prisma
model User {
  id           Int     @id @default(autoincrement())
  name         String?
  writtenPosts Post[]  @relation("WrittenPosts")
  pinnedPost   Post?   @relation("PinnedPost")
}
model Post {
  id          Int     @id @default(autoincrement())
  title       String?
  author      User    @relation("WrittenPosts", fields: [authorId], references: [id])
  authorId    Int
  pinnedBy    User?   @relation(name: "PinnedPost", fields: [pinnedById], references: [id])
  pinnedById  Int?
}
```

Is it possible to create a `User`, with a `Post`, which has the proper value of `authorId`, and when `pinnedBy` links to the newly created `User`?

```js
await prisma.user.create({
  name: 'John Smith',
  writtenPosts: {
    create: {
      title: 'first post for the John',
      pinnedBy: {
        connect: {
          id: ???   // it should be id of newly created user ('John Smith')
        }
      }
    }
  }
});
```

**Comment by ryands17** [comment, marked as answer]
Hey @yarikos 👋 
You do not need to do this `connect` at all. Prisma automatically creates the relation when you use `create` in `writtenPosts`.

```ts
await prisma.user.create({
  name: 'John Smith',
  writtenPosts: {
    create: {
      title: 'first post for the user',
    }
  }
});
```

This is enough to give the author id to the post and an explicit `author` connect is not required.

**Comment by yarikos** [comment]
Thank @ryands17 for your answer.

But I've realized just now, that I've formulated the question in not that way which I meant actually. Sorry, about that. 

The actual question is about creating proper links in both relations.

In this case, it's the new `User`, with new `Post` (when the `User` is an author)  and the `Post` is pinned by the `User`.

I've updated the question.

**Reply by ryands17** [reply]
In that case, separate queries would be the way to go. First create the post, then create the user and connect the post id to both the fields.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
